src/requirements/requirement-graph-builder-from-user-data.ts

```diff
@@ -12,20 +12,26 @@ import buildRequirementFulfillmentGraph from './requirement-graph-builder';
  * helping to compute requirement progress.
  */
 function forfeitTransferCredit(coursesTaken: readonly CourseTaken[]): readonly CourseTaken[] {
+  // filter out AP/IB equivalent courses with legitimate course ids
   const equivalentCourses = coursesTaken.filter(course => course.courseId !== CREDITS_COURSE_ID);
+
+  // generate set for all forfeited equivalent course ids
   const equivalentCourseIds = new Set(equivalentCourses.map(({ courseId }) => courseId));
-  let transferCreditCourses = coursesTaken.filter(course => course.courseId === CREDITS_COURSE_ID);
-  transferCreditCourses = transferCreditCourses.filter(
-    ({ courseId }) => !equivalentCourseIds.has(courseId)
+  // filter out any credits-only courses generated from AP/IB exams
+  const transferCreditCourses = coursesTaken.filter(
+    course => course.courseId === CREDITS_COURSE_ID && !equivalentCourseIds.has(course.courseId)
```

Comment:
A little confused here. What do you mean by credits-only courses? Are those courses that don't apply to any other req except the 120 Academic Credit req? Are you saying that courses with courseid CREDITS_COURSE_ID are transfer credit?

Reply:
I didn't actually change anything in this code, I just slightly simplified it. However, now that I'm looking at it more closely, I'm also slightly confused.

There is a definitely a mistake in there. `equivalentCourseIds` is all the courseIds that are not CREDITS_COURSE_ID. `transferCreditsCourses` as it is right now is all the courseids that are CREDITS_COURSE_ID (the second filter condition is always true). So, this function just returns `coursesTaken`.

I sat here for a while but I'm not satisfied with any of my attempts at an explanation for why this function is/was necessary. I think it has something to do with the nuances of AP/IB credit. Should I take it out for now? Maybe we should take another look next semester (since it's not the topic of this PR)?

Reply:
I think we should take another look next semester. Can you create a Notion Story item on the backlog?

Reply:
https://www.notion.so/courseplan/Explore-forfeitTransferCredit-function-33203aea99b84f03a19c34d4031c4831